dotcom-rendering/src/components/AdSlot.web.tsx

```diff
@@ -66,6 +66,10 @@ type RemainingProps = {
  */
 type Props = DefaultProps & (RightProps | InlineProps | RemainingProps);
 
+type MobileStickyContainerProps = {
+	isValidPageForMobileSticky: boolean;
+};
+
```

Comment:
I thought about making this as an optional parameter but the problem is I am not sure which layouts could have "Article" as a `contentType` or a `pageId` that starts with "football/" that's why I decided to make it mandatory.

Reply:
An alternative would be to pass `pageId` and `contentType` as props the element, and let the decision of whether to be full with to be internal to the component.

Reply:
What I understand you want the logic to happen in the `MobileStickyContainer` and we will just pass to each layout component the `pageId` and `contentType`. I was trying to achieve that but I was able to do it by passing the `article`. Would that be ok? 42760a10346788731fc6944207cbd9b176998afc

I just pushed 2 files to check the change and if it doesn't need more refactoring I will do the same for the rest of the components.

Reply:
Yes – I think that’s fine!

---

As as way to make your code more robust to changes and completely option, you could also tell TypeScript that you specifically only care about `contentType` and `pageId` [using the `Pick<T>` utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys):

```ts
type MobileStickyContainerProps = {
	article: Pick<FEArticleType, 'contentType' | 'pageId'>;
};
```

Reply:
Oh that is interesting! It's the first time I see this utility type. That's what I wanted to do as you suggested but I didn't know how I can implement it. Done in 03feeadd4e9c1ff22255107e8d7e2f1548b93710. Thank you Max!